File: app/src/main/java/com/sharparam/kfs/HomeFragment.java
package com.sharparam.kfs;

import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.sharparam.kfs.api.Page;

import org.json.JSONException;

import java.io.IOException;

import us.feras.mdv.MarkdownView;

public class HomeFragment extends Fragment {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        final View view = inflater.inflate(R.layout.fragment_home, container, false);

        final MarkdownView md = (MarkdownView)view.findViewById(R.id.md_home);

        (new AsyncTask<String, Void, Page>() {
            @Override
            protected Page doInBackground(String... params) {
                try {
                    return Page.findByName(params[0]);
                } catch (IOException e) {
                    return Page.generateErrorPage(e);
                } catch (JSONException e) {
                    return Page.generateErrorPage(e);
                }
            }

            @Override
            protected void onPostExecute(Page page) {
                md.loadMarkdown(page.getContent());
            }
        }).execute("home");

        return view;
    }
}

File: app/src/main/java/com/sharparam/kfs/HomeFragment.kt
package com.sharparam.kfs

import android.os.AsyncTask
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.sharparam.kfs.api.Page
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.support.v4.onUiThread
import org.jetbrains.anko.support.v4.uiThread

import org.json.JSONException

import java.io.IOException

import us.feras.mdv.MarkdownView

class HomeFragment : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater!!.inflate(R.layout.fragment_home, container, false)

        val md = view.findViewById(R.id.md_home) as MarkdownView

        doAsync {
            val page = try {
                Page.findByName("home")
            } catch (e: IOException) {
                Page.generateErrorPage(e)
            } catch (e: JSONException) {
                Page.generateErrorPage(e)
            }

            onUiThread {
                md.loadMarkdown(page.content)
            }
        }

        return view
    }
}
